cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go

```diff
@@ -676,3 +665,24 @@ func installedPackageAvailablePackageRef(rel *helmv2.HelmRelease) (*corev1.Avail
 		Context:    &corev1.Context{Namespace: repoNamespace},
 	}, nil
 }
+
+// ref https://fluxcd.io/docs/components/helm/helmreleases/
+func helmReleaseName(key types.NamespacedName, rel *helmv2.HelmRelease) types.NamespacedName {
+	helmReleaseName := rel.Spec.ReleaseName
+	// according to docs ReleaseName is optional and defaults to a composition of
+	// '[TargetNamespace-]Name'.
+	if helmReleaseName == "" {
+		targetNamespace := rel.Spec.TargetNamespace
+		// according to docs targetNamespace is optional and defaults to the namespace of the HelmRelease
+		if targetNamespace == "" {
+			targetNamespace = key.Namespace
+		}
+		helmReleaseName = fmt.Sprintf("%s-%s", targetNamespace, key.Name)
```

Comment:
Just leaving this here so I can refer to it on your related issue #4189, not suggesting the change for this PR

```suggestion
		helmReleaseName = key.Name
		if targetNamespace != "" {
			helmReleaseName = fmt.Sprintf("%s-%s", targetNamespace, key.Name)
		}
```
which would match the behaviour that I saw by default from flux, I think.

Reply:
ok, sounds good